### PR TITLE
implement timer based console flush

### PR DESCRIPTION
**Pull request by purezen**
Fixes https://github.com/Shippable/cexec/issues/32

### CODE REVIEW

File: script_runner.py [hunk 1]
```diff
@@ -78,6 +79,9 @@ def _run_command(self, cmd, working_dir):
 
         command_thread.start()
 
+        console_flush_thread = threading.Thread(target=self.__trigger_flush_console_output)
```
Comment (rageshkrishna) [inline script_runner.py]: Do we really need a thread here? Why not just create a timer?

Comment (purezen) [inline script_runner.py]: class threading.Timer(interval, function, args=None, kwargs=None)
is the signature of the Timer object.

Comment (rageshkrishna) [inline script_runner.py]: Not sure I understand why that's a problem... why not just do
````
t = threading.Timer(self.config['TIME_INTERVAL'], self.__trigger_flush_console_output)
t.start()
````
like we do from inside the function

Comment (purezen) [inline script_runner.py]: EOD: Initiating __trigger_flush_console ```Timer``` now. However, the build process is now failing. Looking into it.

**Comment by rageshkrishna** [comment]
A couple of lines are probably > 80 char limit but looks good to me otherwise.

@ric03uec Any suggestions on this? We're going to do the same thing in mexec as well.

**Pull request merged**